java-frontend/src/main/java/org/sonar/java/se/NullableAnnotationUtils.java

```diff
@@ -92,11 +119,34 @@ public static boolean isGloballyAnnotatedWith(MethodTree methodTree, String anno
     if (isAnnotatedWith(packageSymbol, annotation)) {
       return valuesForAnnotation(packageSymbol, annotation);
     }
-    return Collections.emptyList();
+    return null;
   }
 
+  @CheckForNull
   private static List<SymbolMetadata.AnnotationValue> valuesForAnnotation(Symbol symbol, String annotation) {
     return symbol.metadata().valuesForAnnotation(annotation);
   }
 
+  private static boolean containsDefaultLocation(Tree defaultLocation, String target) {
+    Symbol symbol;
+    switch (defaultLocation.kind()) {
+      case IDENTIFIER:
+        symbol = ((IdentifierTree) defaultLocation).symbol();
+        break;
+      case MEMBER_SELECT:
+        symbol = ((MemberSelectExpressionTree) defaultLocation).identifier().symbol();
+        break;
+      case NEW_ARRAY:
+        return ((NewArrayTree) defaultLocation).initializers().stream().anyMatch(expr -> containsDefaultLocation(expr, target));
+      default:
+        // other tree which may be used in annotations ?
```

Comment:
this comment is rather odd... I guess the message in the exception is sufficient.